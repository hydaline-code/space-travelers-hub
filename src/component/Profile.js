
import React from 'react';
import { useSelector } from 'react-redux';
import './styles/Profile.css';

const Profile = () => {
  const missions = useSelector((state) => state.missions.missions);
  const dragons = useSelector((state) => state.dragons.dragons);
  const reservedDragons = dragons.filter((dragon) => dragon.reserved !== false);
  const joinedMissions = missions.filter((mission) => mission.reserved);
  const rockets = useSelector((state) => state.rockets.rockets);
  const reservedRockets = useSelector((state) => state.rockets.rockets.filter((rocket) => rocket.reserved));


  return (
    <div className="profile-container">
      <section>
        <div className="Rockets">
          <h2>My Rockets</h2>
          <ul className="reserved-list">
          {reservedRockets.map((rocket) => (
  <li key={rocket.id} className="reserved-item">
    <div className="reserved-details">
      <p>{rocket.rocket_name}</p>
    </div>
  </li>
))}
          </ul>
        </div>
      </section>
      <section>
        <div className="Rockets">
          <h2>My Missions</h2>
          <ul className="reserved-list">
            {joinedMissions.map((mission) => (
              <li key={mission.mission_id} className="reserved-item">
                <p>{mission.mission_name}</p>
              </li>
            ))}
          </ul>
        </div>
      </section>
      <section>
        <div className="Rockets">
          <h2>My Dragons</h2>
          <ul className="reserved-list">
            {reservedDragons.map((dragon) => (
              <li key={dragon.id} className="reserved-item">
                <p>{dragon.name}</p>
              </li>
            ))}
          </ul>
        </div>
      </section>
    </div>
  );
};

export default Profile;

