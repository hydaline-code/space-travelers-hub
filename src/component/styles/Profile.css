.profile-container {
  padding: 1rem;
}

.list-group {
  padding: 0.5rem 0.5rem;
  border-radius: 0;
}

ul > .list-group-item {
  font-weight: bold;
  font-size: 20px;
}
