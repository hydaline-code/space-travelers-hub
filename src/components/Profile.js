import React from 'react';

const Profile = () => (
  <div>My profile</div>
);

export default Profile;
