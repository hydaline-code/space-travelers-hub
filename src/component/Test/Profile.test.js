import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Profile from '../Profile';


const mockStore = configureStore([]);

describe('Profile Component', () => {
  it('matches the snapshot', () => {
    const initialState = {
      missions: {
        missions: [
          {
            mission_id: '1',
            mission_name: 'Mission 1',
            reserved: true,
          },
          {
            mission_id: '2',
            mission_name: 'Mission 2',
            reserved: false,
          },
        ],
      },
      dragons: {
        dragons: [
          {
            id: '1',
            name: 'Dragon 1',
            reserved: true,
          },
          {
            id: '2',
            name: 'Dragon 2',
            reserved: false,
          },
        ],
      },
      rockets: {
        rockets: [
          {
            id: '1',
            rocket_name: 'Rocket 1',
            reserved: true,
          },
          {
            id: '2',
            rocket_name: 'Rocket 2',
            reserved: false,
          },
        ],
      },
    };

    const store = mockStore(initialState);

    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
