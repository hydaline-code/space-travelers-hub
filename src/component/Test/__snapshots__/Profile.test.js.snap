// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component matches the snapshot 1`] = `
<div
  className="profile-container"
>
  <section>
    <div
      className="Rockets"
    >
      <h2>
        My Rockets
      </h2>
      <ul
        className="reserved-list"
      >
        <li
          className="reserved-item"
        >
          <div
            className="reserved-details"
          >
            <p>
              Rocket 1
            </p>
          </div>
        </li>
      </ul>
    </div>
  </section>
  <section>
    <div
      className="Rockets"
    >
      <h2>
        My Missions
      </h2>
      <ul
        className="reserved-list"
      >
        <li
          className="reserved-item"
        >
          <p>
            Mission 1
          </p>
        </li>
      </ul>
    </div>
  </section>
  <section>
    <div
      className="Rockets"
    >
      <h2>
        My Dragons
      </h2>
      <ul
        className="reserved-list"
      >
        <li
          className="reserved-item"
        >
          <p>
            Dragon 1
          </p>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
